### PR TITLE
fix(*): make modifier ProduceAllIfProducesIsEmpty also modify ErrorProduces

### PR DESCRIPTION
**Pull request by knight42**
**What this PR does / why we need it**:

The value of `ErrorProduces` field of `Definition` is the same as `Produces` field by default:
1. https://github.com/caicloud/nirvana/blob/b561a068104cef53a92bd475c326231132ec6158/service/rpc/builder.go#L129
2. https://github.com/caicloud/nirvana/blob/b561a068104cef53a92bd475c326231132ec6158/service/rest/builder.go#L157

Currently the modifier  `ProduceAllIfProducesIsEmpty` only modify the `Produces` field, so `ErrorProduces` would be empty If `Produces` is empty, then nirvana will throw an error when building executor:
https://github.com/caicloud/nirvana/blob/b561a068104cef53a92bd475c326231132ec6158/service/executor/executor.go#L59-L61

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #

Reference to #

**Special notes for your reviewer**:

/cc @iawia002 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: service/modfier_test.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package service
```
Comment (iawia002) [inline service/modfier_test.go:1]: Needs the file header here

Comment (knight42) [reply service/modfier_test.go:17]: fixed

**Pull request merged**